xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-page/src/main/java/org/xwiki/test/page/PageTest.java

```diff
@@ -331,41 +382,13 @@ private void initializeSkinEnvironment() throws ComponentLookupException
     protected void initializeEnvironmentResources() throws Exception
     {
         // Environment resources
-        Environment environment = oldcore.getMocker().getInstance(Environment.class);
-        when(environment.getResource(any(String.class))).thenAnswer(
-            (Answer) invocation -> {
-                String resourceName = (String) invocation.getArguments()[0];
-                // Algorithm:
-                // - Try to load the passed resource name from the CL first
-                // - If the resource is a template (ends in .vm) then remove the /skins/flamingo prefix from the
-                //   resource name and try again to load it from the CL
-                // - If not found, then let tests be able to override {@code getEnvironmentResource()}.
-                URL url = getClass().getResource(resourceName);
-                if (url == null) {
-                    url = getClass().getResource(getShortTemplateResourceName(resourceName));
-                    if (url == null) {
-                        url = getEnvironmentResource(resourceName);
-                    }
-                }
-                return url;
-            });
-        when(environment.getResourceAsStream(any(String.class))).thenAnswer(
-            (Answer) invocation -> {
-                String resourceName = (String) invocation.getArguments()[0];
-                // Algorithm:
-                // - Try to load the passed resource name from the CL first
-                // - If the resource is a template (ends in .vm) then remove the /skins/flamingo prefix from the
-                //   resource name and try again to load it from the CL
-                // - If not found, then let tests be able to override {@code getEnvironmentResource()}.
-                InputStream is = getClass().getResourceAsStream(resourceName);
-                if (is == null) {
-                    is = getClass().getResourceAsStream(getShortTemplateResourceName(resourceName));
-                    if (is == null) {
-                        is = getEnvironmentResourceAsStream(resourceName);
-                    }
-                }
-                return is;
-            });
+        Environment environment = this.oldcore.getMocker().getInstance(Environment.class);
+        when(environment.getResource(any(String.class))).thenAnswer(new GetResourceAnswer());
+        when(environment.getResourceAsStream(any(String.class))).thenAnswer(new GetResourceAsStreamAnswer());
+
+        XWikiEngineContext engineContext = mock(XWikiEngineContext.class);
+        when(engineContext.getResourceAsStream(any())).thenAnswer(new GetResourceAsStreamAnswer());
+        this.xwiki.setEngineContext(engineContext);
```

Comment:
`Answer<?>` anonymous classes moved to nested classes at the top of the class. 
`GetResourceAsStreamAnswer` reused for the class to `getResourceAsStream` from xwiki engine context.